src/index.js

```diff
@@ -505,28 +535,49 @@ function extendReactElementWithRef(path) {
   if (!typeIdentifierNode || !propsIdentifierNode) {
     return;
   }
-  const typeIdentifierValueIsIdentifier = t.isIdentifier(typeIdentifierNode.value);
-  // variable "type" is a MemberExpression in production code
-  const typeIdentifierValueIsMemberExpression = t.isMemberExpression(typeIdentifierNode.value);
 
-  if (
-    !(typeIdentifierValueIsIdentifier || typeIdentifierValueIsMemberExpression) ||
-    !t.isIdentifier(propsIdentifierNode.value)
-  ) {
+  if (!t.isIdentifier(propsIdentifierNode.value)) {
     return;
   }
-  const typeIdentifier = typeIdentifierValueIsIdentifier
-    ? typeIdentifierNode.value.name
-    : typeIdentifierNode.value.object.name;
   const propsIdentifier = propsIdentifierNode.value.name;
   const refIdentifier = refIdentifierNode && refIdentifierNode.value.name;
 
   // pass props, type, and ref values to the createFabricRefCode for processing
   // again, these variable identifiers are dynamic due to minification
-  const fabricRefCodeAST = babylon.parse(
-    _createFabricRefCode(refIdentifier, typeIdentifier, propsIdentifier),
-  );
-  path.getStatementParent().insertBefore(fabricRefCodeAST.program.body);
+  const fabricRefCodeAST = babylon.parse(_createFabricRefCode(refIdentifier, propsIdentifier));
+
+  // Insert the function within the function body depending on the type of declaration
+  if (refIdentifier) {
+    // React versions < 19
+    path.getStatementParent().insertBefore(fabricRefCodeAST.program.body);
```

Comment:
React versions prior to 19 will receive our rewritten function in the same location, but for versions greater than 19, it will be at the beginning of the function due to the location of `config = maybeKey.ref` and `self = props.ref`.